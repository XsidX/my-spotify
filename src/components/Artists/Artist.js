import React from 'react';
import { Link } from 'react-router-dom';

import PropTypes from 'prop-types';

const Artist = ({ artist }) => {
  const { id, name, images } = artist;

  return (
    <Link to={`/artist/${id}`} className="relative h-48 col-span-1">
      <img src={images[0]?.url} alt={name} className="object-contain" />

      <div className="absolute bottom-2 left-4 text-white">
        <h2 className="text-sm font-medium">{name}</h2>
      </div>
    </Link>
  );
};

Artist.propTypes = {
  artist: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    images: PropTypes.arrayOf(
      PropTypes.shape({
        url: PropTypes.string.isRequired,
      }),
    ).isRequired,
    followers: PropTypes.shape({
      total: PropTypes.number.isRequired,
    }).isRequired,
  }).isRequired,
};
export default Artist;
