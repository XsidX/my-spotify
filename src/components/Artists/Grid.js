import React from 'react';
import PropTypes from 'prop-types';

const Grid = ({ children }) => (
  <div className="grid grid-cols-2 md:grid-cols-4">{children}</div>
);

Grid.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Grid;
