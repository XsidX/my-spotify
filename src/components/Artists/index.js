import React from 'react';
import { useSelector } from 'react-redux';
import Artist from './Artist';
import Grid from './Grid';

const Artists = () => {
  const artists = useSelector((state) => state.artists.artists) || [];
  return (
    <Grid>
      {artists.map((artist) => (
        <Artist key={artist.id} artist={artist} />
      ))}
    </Grid>
  );
};

export default Artists;
