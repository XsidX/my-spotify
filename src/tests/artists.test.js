import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';
import artistsReducer from '../store/reducers/artists-slice';
import albumsReducer from '../store/reducers/albums-slice';
import uiReducer from '../store/reducers/ui-slice';
import Artists from '../components/Artists';

const initialState = {
  artists: [],
  albums: [],
  ui: {
    toolBarIsIntersecting: false,
  },
};

const store = configureStore({
  reducer: {
    artists: artistsReducer,
    albums: albumsReducer,
    ui: uiReducer,
  },
  preloadedState: initialState,
});

describe('Artists', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Artists />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
