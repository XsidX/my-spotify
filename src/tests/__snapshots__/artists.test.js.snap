// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists renders correctly 1`] = `
<div
  className="grid grid-cols-2 md:grid-cols-4 bg-black"
/>
`;
